packages/grafana-ui/src/components/Table/TableNG/components/TableCellTooltip.tsx

```diff
@@ -0,0 +1,159 @@
+
+import { Popover } from '../../../Tooltip/Popover';
+import { TableCellOptions } from '../../types';
+import { getTooltipStyles } from '../styles';
+import { TableCellRenderer, TableCellRendererProps } from '../types';
+
+export interface Props {
+  cellOptions: TableCellOptions;
+  children: ReactElement;
+  classes: ReturnType<typeof getTooltipStyles>;
+  className?: string;
+  data: DataFrame;
+  disableSanitizeHtml?: boolean;
+  field: Field;
+  getActions: (field: Field, rowIdx: number) => ActionModel[];
+  gridRef: RefObject<DataGridHandle>;
+  height: number;
+  placement?: TableCellTooltipPlacement;
+  renderer: TableCellRenderer;
+  rowIdx: number;
+  style?: CSSProperties;
+  tooltipField: Field;
+  theme: GrafanaTheme2;
+  width?: number;
+}
+
+export function TableCellTooltip({
+  cellOptions,
+  children,
+  classes,
+  className,
+  data,
+  disableSanitizeHtml,
+  field,
+  getActions,
+  gridRef,
+  height,
+  placement,
+  renderer,
+  rowIdx,
+  style,
+  theme,
+  tooltipField,
+  width = 300,
+}: Props) {
+  const rawValue = field.values[rowIdx];
+  const tooltipCaretRef = useRef<HTMLDivElement>(null);
+
+  const [hovered, setHovered] = useState(false);
+  const [pinned, setPinned] = useState(false);
+
+  const show = hovered || pinned;
+  const dynamicHeight = tooltipField.config.custom?.cellOptions?.dynamicHeight;
+
+  useEffect(() => {
+    if (pinned) {
+      const gridRoot = gridRef.current?.element;
+
+      const windowListener = (ev: Event) => {
+        if (ev.target === tooltipCaretRef.current) {
+          return;
+        }
+
+        setPinned(false);
+        window.removeEventListener('click', windowListener);
+      };
+
+      window.addEventListener('click', windowListener);
+
+      // right now, we kill the pinned tooltip on any form of scrolling to avoid awkward rendering
+      // where the tooltip bumps up against the edge of the scrollable container. we could try to
+      // kill the tooltip when it hits these boundaries rather than when scrolling starts.
+      const scrollListener = () => {
+        setPinned(false);
+      };
+      gridRoot?.addEventListener('scroll', scrollListener, { once: true });
+
+      return () => {
+        window.removeEventListener('click', windowListener);
+        gridRoot?.removeEventListener('scroll', scrollListener);
+      };
+    }
+
+    return;
+  }, [pinned, gridRef]);
+
+  const rendererProps = useMemo(
+    () =>
+      ({
+        cellInspect: false,
+        cellOptions,
+        disableSanitizeHtml,
+        field,
+        frame: data,
+        getActions,
+        height,
+        rowIdx,
+        showFilters: false,
+        theme,
+        value: rawValue,
+        width,
+      }) satisfies TableCellRendererProps,
+    [cellOptions, data, disableSanitizeHtml, field, getActions, height, rawValue, rowIdx, theme, width]
+  );
+
+  const cellElement = tooltipCaretRef.current?.closest<HTMLElement>('.rdg-cell');
```

Comment:
might be better to populate this from inside the event handlers instead of running the selector eagerly for all rendered cells?

Reply:
It's used as a required prop of `Popover`.

Reply:
does popover need to be exist before there is an interaction tho?

Reply:
done! I have to stick it in a ref to access it ad-hoc like this, but looking it up JIT is worth it.